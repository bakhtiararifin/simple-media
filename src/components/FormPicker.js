import React from 'react'
import { View, Picker } from 'react-native'
import { FormLabel, Divider } from 'react-native-elements'

const CustomFormPicker = ({ label, selectedValue, options = [], style = {}, ...otherProps }) => {
  const defaultStyle = {
    marginHorizontal: 15,
  }

  return (
    <View>
      {label && <FormLabel>{label}</FormLabel>}
      <Picker selectedValue={selectedValue} style={{ ...defaultStyle, ...style }} {...otherProps}>
        {options.map(option => (
          <Picker.Item key={option.value} label={option.label} value={option.value} />
        ))}
      </Picker>
      <Divider style={{ marginHorizontal: 15 }} />
    </View>
  )
}

export default CustomFormPicker
