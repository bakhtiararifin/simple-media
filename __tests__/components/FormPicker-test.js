import React from 'react'
import FormPicker from '@components/FormPicker'

import renderer from 'react-test-renderer'

test('renders correctly', () => {
  const tree = renderer
    .create(
      <FormPicker
        label="Gender"
        selectedValue="male"
        options={[{ label: 'Male', value: 'male' }, { label: 'Female', value: 'female' }]}
      />,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders correctly without label', () => {
  const tree = renderer
    .create(
      <FormPicker
        selectedValue="male"
        options={[{ label: 'Male', value: 'male' }, { label: 'Female', value: 'female' }]}
      />,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
