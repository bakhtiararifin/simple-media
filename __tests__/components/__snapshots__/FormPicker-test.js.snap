// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      Gender
    </Text>
  </View>
  <View
    style={
      Object {
        "marginHorizontal": 15,
      }
    }
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "Male",
            "textColor": undefined,
            "value": "male",
          },
          Object {
            "label": "Female",
            "textColor": undefined,
            "value": "female",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#e1e8ee",
          "height": 1,
        },
        Object {
          "marginHorizontal": 15,
        },
      ]
    }
  />
</View>
`;

exports[`renders correctly without label 1`] = `
<View>
  <View
    style={
      Object {
        "marginHorizontal": 15,
      }
    }
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "Male",
            "textColor": undefined,
            "value": "male",
          },
          Object {
            "label": "Female",
            "textColor": undefined,
            "value": "female",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#e1e8ee",
          "height": 1,
        },
        Object {
          "marginHorizontal": 15,
        },
      ]
    }
  />
</View>
`;
